src/main/java/hw02/FibonacciAlgorithms.java

```diff
@@ -0,0 +1,67 @@
+package hw02;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class FibonacciAlgorithms {
+    /**
+     * Recursive implementation of Fibonacci sequence
+     * Time Complexity: exponential, each time num of operations is 2^n where n is depth of tree of calls
+     * Space Complexity: linear, each time we append one frame in recursion stack of calls
+     * <p>
+     * Explanation: Each call branches into two recursive calls, creating
+     * a binary tree of calls with height n. The same subproblems are
+     * solved multiple times, leading to exponential time complexity.
+     */
+    public static long fibonacciRecursive(int n) {
+        if (n < 0) throw new IllegalArgumentException("n should be grater than 0");
+        else if (n <= 1) return n;
+        else return fibonacciRecursive(n - 1) + fibonacciRecursive(n - 2);
+    }
+
+    /**
+     * Memoized implementation of Fibonacci sequence
+     * Time Complexity: linear, compute each new level of tree calls just once
+     * Space Complexity: linear, preserve each fibonacci number into cache and each frame in recursion stack of calls
+     * <p>
+     * Explanation: By caching intermediate results, we avoid
+     * redundant calculations. Each number from 0 to n is
+     * calculated exactly once.
+     */
+    public static HashMap<Integer, Long> cache = new HashMap<>();
```

Comment:
having mutable static state is normally should be avoided if there isn't a very good reason to keep it.
acceptable for exercise though